addon/addon/models/sprite-snapshot-node-builder.ts

```diff
@@ -307,16 +307,53 @@ export class SpriteSnapshotNodeBuilder {
 
     // Classify non-natural KeptSprites
     for (let insertedSpriteModifier of classifiedInsertedSpriteModifiers) {
+      let counterpartSpriteModifier: SpriteModifier | undefined;
+
+      // find a suitable RemovedSprite counterpart if any
+      let removedSpriteModifiers = [...classifiedRemovedSpriteModifiers].filter(
+        (removedSpriteModifier) =>
+          new SpriteIdentifier(
+            insertedSpriteModifier.id,
+            insertedSpriteModifier.role
+          ).equals(
+            new SpriteIdentifier(
+              removedSpriteModifier.id,
+              removedSpriteModifier.role
+            )
+          )
+      );
+
+      assert(
+        'Multiple matching removedSpriteModifiers found',
+        removedSpriteModifiers.length < 2
+      );
+
+      let removedSpriteModifier = removedSpriteModifiers[0] as SpriteModifier;
+      if (removedSpriteModifier) {
+        classifiedRemovedSpriteModifiers.delete(removedSpriteModifier);
```

Comment:
One use case I can think of for a removed sprite is keeping a "phantom/greyed out version" while the inserted sprite moves towards its new position +- does element transitions. If an intermediate sprite is introduced in the middle of this and takes over the role of counterpart, we no longer have this ability if the removed sprite is removed from this set. 

Is there a good way to express a current render counterpart (removed sprite) differently from a previous render counterpart (kept sprite that was interrupted), and should we consider raising this issue w Ed?